pkg/catalog/catalog.go

```diff
@@ -1,18 +1,114 @@
 package catalog
 
 import (
+	"encoding/json"
 	"fmt"
+	// "io/ioutil"
+	// "net/http"
 	"sort"
 	"strings"
+	// "time"
 
 	"github.com/golang/glog"
 	imagev1 "github.com/openshift/api/image/v1"
 	"github.com/openshift/odo/pkg/occlient"
 	"github.com/openshift/odo/pkg/util"
 	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// CheDevfileRegistry is Che devfile registry reposotiry
+const CheDevfileRegistry = "https://che-devfile-registry.openshift.io/"
+
+// DevfileIndexLink is the index.json link of Che devfile registry
+const DevfileIndexLink = CheDevfileRegistry + "/devfiles/index.json"
+
+// ListDevfileComponents lists all the available devfile components
+func ListDevfileComponents() (DevfileComponentTypeList, error) {
+
+	var devfileIndex []DevfileIndexEntry
+	var catalogDevfileList DevfileComponentTypeList
+	catalogDevfileList.DevfileRegistry = CheDevfileRegistry
+
+	jsonBytes, err := util.HTTPGetRequest(DevfileIndexLink)
+	if err != nil {
+		return DevfileComponentTypeList{}, fmt.Errorf("Unable to download devfile index.json from %s", DevfileIndexLink)
+	}
+
+	err = json.Unmarshal(jsonBytes, &devfileIndex)
+
+	for _, devfileIndexEntry := range devfileIndex {
+		devfileIndexEntryName := devfileIndexEntry.DisplayName
+		devfileIndexEntryDescription := devfileIndexEntry.Description
+		devfileIndexEntryLink := devfileIndexEntry.Links.Link
+
+		var yamlBytes []byte
+		
+		devfileLink := CheDevfileRegistry + devfileIndexEntryLink
+		yamlBytes, err = util.HTTPGetRequest(devfileLink)
```

Comment:
This part worries me a bit. Downloading all devfiles can get time-consuming for bigger registries.
I know that we can't do much about it right now. Just pointing this out so we can think about it.